tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaListenerNameTest.java

```diff
@@ -61,4 +65,47 @@ public void testListenerName() throws Exception {
 
         super.internalCleanup();
     }
+
+    @Test(timeOut = 30000)
+    public void testMultipleListenerName() throws Exception {
+        super.resetConfig();
+        conf.setAdvertisedAddress(null);
+        final String localAddress = ServiceConfigurationUtils.getDefaultOrConfiguredAddress(null);
+        conf.setInternalListenerName("pulsar");
+        final String kafkaProtocolMap = "kafka:PLAINTEXT,kafka_external:PLAINTEXT";
+        conf.setKafkaProtocolMap(kafkaProtocolMap);
+        int externalPort = PortManager.nextFreePort();
+        final String kafkaListeners = "kafka://0.0.0.0:" + kafkaBrokerPort
+                + ",kafka_external://0.0.0.0:" + externalPort;
+        conf.setKafkaListeners(kafkaListeners);
+        final String advertisedListeners =
+                "pulsar:pulsar://" + localAddress + ":" + brokerPort
+                        + ",kafka:pulsar://" + "localhost:" + kafkaBrokerPort
+                        + ",kafka_external:pulsar://" + "localhost:" + externalPort;
+        conf.setAdvertisedListeners(advertisedListeners);
+        log.info("Set advertisedListeners to {}", advertisedListeners);
+        super.internalSetup();
+
+        kafkaProducerSend("localhost:" + kafkaBrokerPort);
+        kafkaProducerSend("localhost:" + externalPort);
+
+        super.internalCleanup();
+    }
+
+    private void kafkaProducerSend(String server) {
+        final Properties props = new Properties();
+        props.setProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, server);
+        props.setProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        props.setProperty(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+
+        final KafkaProducer<String, String> producer = new KafkaProducer<>(props);
+        producer.send(new ProducerRecord<>("my-topic", "my-message"), (metadata, exception) -> {
+            if (exception == null) {
+                log.info("Send to {}", metadata);
+            } else {
+                log.error("Send failed: {}", exception.getMessage());
+            }
+        });
```

Comment:
We should use a synchronous send here to propagate the exception if send failed. Here it call `send()`, which is asynchronous, and then close the producer immediately, it cannot guarantee the send is completed.

Reply:
A simple fix is

```java
producer.send(new ProducerRecord<>("my-topic", "my-message")).get();
```

if you don't want to print the logs.

Reply:
done